src/format.ts

```diff
@@ -110,8 +120,117 @@ function checkFormat(srcFiles: string[], baseArgs: string[]): Promise<boolean> {
     out.on('data', (data: Buffer) => {
       output += data;
     });
+
     out.on('end', () => {
-      resolve(output.indexOf('<replacement ') === -1 ? true : false);
+      findFormatErrorLines(output, options).then(() => {
+        resolve(output.indexOf('<replacement ') === -1 ? true : false);
+      });
+    });
+  });
+}
+
+/**
+ * Parses through xml string for the replacement offsets and lengths. Uses those
+ * values to locate the formatting error lines.
+ *
+ * @param output xml string
+ * @param options
+ */
+async function findFormatErrorLines(output: string, options: Options) {
+  const files = output.split('<?xml version=\'1.0\'?>\n');
+
+  for (let i = 1; i < files.length; i++) {
+    const formatErr = {
+      offset: output.match(/(?<=offset=\')(\d+)(?=\')/g),
+      length: output.match(/(?<=length=\')(\d+)(?=\')/g),
+      fix: output.match(/(?<=length=\'\d+\'>)(.*)(?=<\/replacement>)/g)
+    };
+
+    if (formatErr.length === null || formatErr.offset === null ||
+        formatErr.fix === null) {
+      return;
```

Comment:
So, instead of checking if the variables are null, should I check if they are not null and using continue ...and have an else block with the print statement that tells the user that there are no formatting issues? The only case that I can think of where formatErr.length, offset, and fix would be null, is if there are no formatting issues. 

```
if (formatErr.length !== null && formatErr.offset !== null &&
    formatErr.fix !== null) {
  continue;
}else{
  options.logger.log('clang-format reported no errors')
}
```

Reply:
Take a look at the changes I made so far. I'm returning an empty string[] instead of undefined. I don't think it's necessary to print to stderr because it is not an error when formatErr.length/offset/fix is null. it means that the format of the file is fine and does not need to be changed.

Reply:
> The only case that I can think of where formatErr.length, offset, and fix would be null, is if there are no formatting issues.

Yep, I forgot that it's possible for there to not be replacements at all, indicating no formatting issues. For handling this, you can consider `continue`-ing earlier, even before the regexes are being executed, by checking whether the `replacement` tag occurs at all. That way, when you get to the point where `formatErr.length === null || formatErr.offset === null || formatErr.fix === null` is being evaluated, you know that if that statement is true, it indicates that `clang-format` didn't emit expected output. Even though I wouldn't expect `clang-format` to change its output xml format in the future, this would help us disambiguate whether we are continuing because there are no formatting issues, or because an unexpected change occurred in `clang-format`'s output.

Reply:
done